var/spack/repos/builtin/packages/py-chainer/package.py

```diff
@@ -34,3 +37,50 @@ class PyChainer(PythonPackage):
     depends_on('py-filelock', type=('build', 'run'))
     depends_on('py-protobuf@3:', type=('build', 'run'))
     depends_on('py-typing@:3.6.6', when='@:6', type=('build', 'run'))
+
+    # Dependencies only required for test of ChainerMN
+    depends_on('py-matplotlib', type=('build', 'run'), when='+mn')
+    depends_on('py-mpi4py', type=('build', 'run'), when='+mn')
+    depends_on("mpi", type=("build", "run"), when='+mn')
+
+    @run_after('install')
+    def cache_test_sources(self):
+        if '+mn' in self.spec:
+            self.cache_extra_test_sources("examples")
+
+    def test(self):
+        if "+mn" in self.spec:
+            # Run test of ChainerMN
+            test_dir = self.test_suite.current_test_data_dir
+            exe_name = join_path(test_dir, "run-test.sh")
+
+            mnist_dir = join_path(
+                self.install_test_root, "examples", "chainermn", "mnist"
+            )
+            mnist_file = join_path(mnist_dir, "train_mnist.py")
+            mpi_name = self.spec["mpi"].prefix.bin.mpirun
+            python_exe = self.spec["python"].command.path
+            opts = [
+                exe_name,
+                mpi_name,
+                "-n",
+                "4",
+                python_exe,
+                mnist_file,
+                "-o",
+                test_dir,
+            ]
+            env["OMP_NUM_THREADS"] = "4"
+            lib_path = join_path(
+                "/usr", "lib", "FJSVtcs", "ple", "lib64", "libpmix.so"
+            )
```

Comment:
This doesn't look like it would work on any other machine. Could we add a dependency for this and register it as an external package?

Reply:
@ketsubouchi will re-submit this PR.